### PR TITLE
Widget inserter: Clarify that the button toggles the inserter

### PR DESCRIPTION
**Pull request by walbo**
## Description
Follow-up on #29759
Add the same functionality to the inserter button on the widget screen and rename the label.

## How has this been tested?
Locally by toggling the inserter sidebar.

## Screenshots 
widget-inserter

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by Mamaduka — APPROVED** [review]
This looks good. Thanks, @walbo!

**Comment by draganescu** [comment]
👋🏻 how come we need to implement this in this specific editor? Isn't that application header common accross all editors?

**Comment by walbo** [comment]
> 👋🏻 how come we need to implement this in this specific editor? Isn't that application header common accross all editors?

Seems like the same functionality is implementet both in `edit-post` and `edit-site` package. As you say these should probably be aligned in a common component.

I can create a alternative PR later today that explores shared code.

**Comment by Mamaduka** [comment]
I had a similar question about the different component a few months ago - https://github.com/WordPress/gutenberg/issues/31597#issuecomment-834348003.

TL;DR - Sometimes duplication is better than bad early abstraction.
